velox/functions/prestosql/JsonFunctions.cpp

```diff
@@ -130,45 +241,114 @@ class JsonParseFunction : public exec::VectorFunction {
   }
 
  private:
-  simdjson::error_code parse(size_t size) const {
+  simdjson::error_code parse(size_t size, std::vector<StringView>& jsonViews)
+      const {
     simdjson::padded_string_view paddedInput(
         paddedInput_.data(), size, paddedInput_.size());
     SIMDJSON_ASSIGN_OR_RAISE(auto doc, simdjsonParse(paddedInput));
-    SIMDJSON_TRY(validate<simdjson::ondemand::document&>(doc));
+    SIMDJSON_TRY(validate<simdjson::ondemand::document&>(doc, jsonViews));
     if (!doc.at_end()) {
       return simdjson::TRAILING_CONTENT;
     }
     return simdjson::SUCCESS;
   }
 
   template <typename T>
-  static simdjson::error_code validate(T value) {
+  static simdjson::error_code validate(
+      T value,
+      std::vector<StringView>& jsonViews) {
     SIMDJSON_ASSIGN_OR_RAISE(auto type, value.type());
     switch (type) {
       case simdjson::ondemand::json_type::array: {
         SIMDJSON_ASSIGN_OR_RAISE(auto array, value.get_array());
+
+        jsonViews.push_back(kArrayStart);
         for (auto elementOrError : array) {
           SIMDJSON_ASSIGN_OR_RAISE(auto element, elementOrError);
-          SIMDJSON_TRY(validate(element));
+          std::vector<StringView> arrayElement;
+          SIMDJSON_TRY(validate(element, arrayElement));
+          jsonViews.insert(
+              jsonViews.end(),
+              std::make_move_iterator(arrayElement.begin()),
+              std::make_move_iterator(arrayElement.end()));
+          jsonViews.push_back(kSeparator);
         }
+
+        // Remove last separator.
+        jsonViews.pop_back();
+        jsonViews.push_back(kArrayEnd);
+
         return simdjson::SUCCESS;
       }
+
       case simdjson::ondemand::json_type::object: {
         SIMDJSON_ASSIGN_OR_RAISE(auto object, value.get_object());
+
+        std::vector<std::pair<StringView, std::vector<StringView>>> objFields;
```

Comment:
We ended up creating a lot `std::vector<StringView>` in this way.  It would nice if we can append them all in the current `jsonViews` and reorder them in-place.

Reply:
My concern is that the re-order itself will not be that cheap since the range of string views being swapped will be of different sizes . Let me still try it out though..